probes/freeOfUntrustedBinaryArtifacts/def.yml

```diff
@@ -0,0 +1,28 @@
+# Copyright 2023 OpenSSF Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+id: freeOfUntrustedBinaryArtifacts
+short: Checks if the project has binary files in its source tree. The probe skips trusted binary files which currently are gradle-wrappers.
+motivation: >
+  Binary files are not readable so users can't see what they do. Many programming language systems can generate executables from source code (e.g., C/C++ generated machine code, Java .class files, Python .pyc files, and minified JavaScript). Users will often directly use executables if they are included in the source repository, leading to many dangerous behaviors.
+implementation: >
+  The implementation looks for the presence of binary files. This is a more permissive probe than "freeOfAnyBinaryArtifacts" which does not trust any binary files.
```

Comment:
nit: `The implementation looks for the presence of binary files that are not "verified". A verified binary is one that ...`

Reply:
Done in https://github.com/ossf/scorecard/pull/3508/commits/c7b8915f14f4db04664e220b9cb3f3ec2affa4e3